### PR TITLE
Fixes several missing-prototype warnings.

### LINKED ISSUE #164: Several missing-prototype warnings
Compiling under gcc 10.3.1 and Linux 6.9.9 generates several missing-prototype warnings.

### PR DESCRIPTION
**Pull request by rountree**
Compiling under Linux 6.9.9 and gcc 10.3.1.

Fixes #164

### CODE REVIEW

File: msr-smp.c
Comment (slabasan) [inline msr-smp.c:1]: Missing license header

File: msr-smp.h
Comment (slabasan) [inline msr-smp.h:1]: Add new lines, remove empty line 5:
```
#ifndef MSR_SMP_INCLUDE
#define MSR_SMP_INCLUDE

int msr_safe_batch(struct msr_batch_array *oa);

#endif
```

File: msr_allowlist.c
Comment (slabasan) [inline msr_allowlist.c:1]: Add new line between standard and local headers

File: msr_entry.c [hunk 1]
```diff
@@ -44,6 +44,7 @@
 #include "msr_batch.h"
 #include "msr_allowlist.h"
 #include "msr_version.h"
+#include "msr_version.h"
```
Comment (slabasan) [inline msr_entry.c:47]: Remove duplicate header `msr_version.h`

File: msr_version.c [hunk 2]
```diff
@@ -11,6 +11,7 @@
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/uaccess.h>
+#include "msr_version.h"
```
Comment (slabasan) [inline msr_version.c:14]: Add new line between standard and local header includes

**Review by slabasan — CHANGES_REQUESTED** [review]
Minor changes

File: msr-smp.c
Comment (rountree) [reply msr-smp.c:1]: Can you check this again?  I'm seeing that `msr-smp.c` has the following at the top of the file:
```
// Copyright 2011-2021 Lawrence Livermore National Security, LLC and other
// msr-safe Project Developers. See the top-level COPYRIGHT file for
// details.
//
// SPDX-License-Identifier: GPL-2.0-only
```

File: msr-smp.h
Comment (rountree) [reply msr-smp.h:1]: Fixed.

File: msr_allowlist.c
Comment (rountree) [reply msr_allowlist.c:1]: Fixed.

File: msr_entry.c (hunk 1 above)
Comment (rountree) [reply msr_entry.c:47]: Fixed.

File: msr_version.c (hunk 2 above)
Comment (rountree) [reply msr_version.c:14]: Fixed.

File: msr-smp.h
Comment (slabasan) [inline msr-smp.h:1]: License missing from this file, not msr-smp.c. Sorry for the confusion!

Comment (rountree) [reply msr-smp.h:1]: Fixed.

**Comment by rountree** [comment]
@slabasan Let me know if you'd like me to squash all of these (or you can do it if you like).

**Review by slabasan — APPROVED** [review]

**Pull request merged**